### PR TITLE
Update types, bump pipiline, externalDns, kube-api-auth

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: scripts/istio-images-mirror [hunk 1]
```diff
@@ -25,8 +25,14 @@ library/nginx rancher/nginx 1.17.4-alpine
...
-quay.io/calico/pod2daemon-flexvol rancher/calico-pod2daemon-flexvol v3.10.2
-us.gcr.io/k8s-artifacts-prod/external-dns/external-dns rancher/kubernetes-external-dns v0.6.0
\ No newline at end of file
+quay.io/calico/node rancher/calico-node v3.12.0
+quay.io/calico/cni rancher/calico-cni v3.12.0
+quay.io/calico/ctl rancher/calico-ctl v3.12.0
+quay.io/calico/kube-controllers rancher/calico-kube-controllers v3.12.0
+quay.io/calico/pod2daemon-flexvol rancher/calico-pod2daemon-flexvol v3.12.0
+us.gcr.io/k8s-artifacts-prod/external-dns/external-dns rancher/kubernetes-external-dns v0.6.0
+gcr.io/google_containers/k8s-dns-kube-dns rancher/k8s-dns-kube-dns 1.15.9
```
Comment (deniseschannon) [inline scripts/istio-images-mirror]: I don't know if this mirror image does mult-arch builds and this image seems to have multi-arch images? 
https://hub.docker.com/repository/docker/rancher/k8s-dns-kube-dns/tags?page=1

File: scripts/istio-images-mirror [hunk 2]
```diff
@@ -25,8 +25,14 @@ library/nginx rancher/nginx 1.17.4-alpine
...
+gcr.io/google_containers/k8s-dns-dnsmasq-nanny rancher/k8s-dns-dnsmasq-nanny 1.15.9
```
Comment (deniseschannon) [inline scripts/istio-images-mirror]: same multi-arch concern: https://hub.docker.com/repository/docker/rancher/k8s-dns-dnsmasq-nanny/tags?page=1

File: scripts/istio-images-mirror [hunk 3]
```diff
@@ -25,8 +25,14 @@ library/nginx rancher/nginx 1.17.4-alpine
...
+gcr.io/google_containers/k8s-dns-sidecar rancher/k8s-dns-sidecar 1.15.9
```
Comment (deniseschannon) [inline scripts/istio-images-mirror]: same multi-arch concern: https://hub.docker.com/repository/docker/rancher/k8s-dns-sidecar/tags?page=1

File: scripts/istio-images-mirror [hunk 4]
```diff
@@ -25,8 +25,14 @@ library/nginx rancher/nginx 1.17.4-alpine
...
+weaveworks/weave-kube rancher/weaveworks-weave-kube 2.6.0
```
Comment (deniseschannon) [inline scripts/istio-images-mirror]: Where are the weavework images used? It doesn't seem to be in Dockerhub?

File: scripts/istio-images-mirror (hunk 1 above)
Comment (StrongMonkey) [reply scripts/istio-images-mirror]: yes this appears to be a problem. I don't think image mirror scripts supports tagging with manifest. I will remove these image then.

File: scripts/istio-images-mirror (hunk 4 above)
Comment (StrongMonkey) [reply scripts/istio-images-mirror]: weaveworks is never used under rancher repo. Here is the issue: https://github.com/rancher/rancher/issues/22954

**Comment by StrongMonkey** [comment]
This PR is updated so that addressing image mirroring will be in a separate PR. This PR will have changes regarding to pipeline,kube-api-auth and globalDNS app update.

**Pull request merged**